internal/beater/java_attacher/java_attacher.go

```diff
@@ -167,6 +177,71 @@ func (j *JavaAttacher) Run(ctx context.Context) error {
 	}
 }
 
+func (j *JavaAttacher) deleteTempDir(tempDir string) {
+	func(name string) {
+		err := os.RemoveAll(name)
+		if err != nil {
+			j.logger.Errorf("failed to delete Java attacher temp dir at %v: %v", tempDir, err)
+		}
+	}(tempDir)
+}
+
+// Creates a temp dir for the java attacher jar with the attacher jar in it, both with 0755 access permissions.
+// The assumption is that the default temp dir will be readable for all users.
+// However, there may be systems in which the temp directory is user-specific,
+// in which case we'll need to modify this function to run a temp dir search per user,
+// maybe by externally executing `echo XXX` and reading the output, relying on environment
+// variables as implemented in os.TempDir, or looking in known paths.
+// In such case, we would switch to using a cache of temp-agent-dir per user.
+func (j *JavaAttacher) createAttacherTempDir() (string, error) {
```

Comment:
I don't want exclusive access, the attacher is ran in a separate process by another user. In addition, as said above, the plan is to generically manage temp-dir per user with proper write permissions, so it will be a similar structure anyway

Reply:
See below

Reply:
This was necessary so that we can cache agent jars instead of downloading every time, and in order to be consistent with what we do in Windows. However, since we learned that we may not be required to have a tmp dir per user in Windows, I will apply this suggestion and we will live without the performance gain of caching agent jars

Reply:
Implemented tmp dir per user